cmd/describe/service/cmd.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright (c) 2020 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package service
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/rosa/pkg/logging"
+	"github.com/openshift/rosa/pkg/ocm"
+	rprtr "github.com/openshift/rosa/pkg/reporter"
+)
+
+var args ocm.DescribeManagedServiceArgs
+
+var Cmd = &cobra.Command{
+	Use:     "service",
+	Aliases: []string{"appliance"},
+	Short:   "Show details of a service",
+	Long:    "Show details of a service",
+	Example: `  # Describe a service with id aaabbbccc"
+  rosa describe service --id=aaabbbccc`,
+	Run:    run,
+	Hidden: true,
+}
+
+func init() {
+	flags := Cmd.Flags()
+
+	flags.StringVar(
+		&args.ID,
+		"id",
+		"",
+		"The id of the service to describe",
+	)
+}
+
+func run(cmd *cobra.Command, argv []string) {
+	reporter := rprtr.CreateReporterOrExit()
+	logger := logging.CreateLoggerOrExit(reporter)
+
+	// Create the client for the OCM API:
+	ocmClient, err := ocm.NewClient().
+		Logger(logger).
+		Build()
+
+	if err != nil {
+		reporter.Errorf("Failed to create OCM connection: %v", err)
+		os.Exit(1)
+	}
+	defer func() {
+		err = ocmClient.Close()
+		if err != nil {
+			reporter.Errorf("Failed to close OCM connection: %v", err)
+		}
+	}()
+
+	// Try to find the cluster:
+	reporter.Debugf("Loading service with id %q", args.ID)
+	service, err := ocmClient.GetManagedService(args)
+	if err != nil {
+		reporter.Errorf("Failed to get service with id %q: %v", args.ID, err)
+		os.Exit(1)
+	}
+
+	fmt.Printf(`
+  %20s: %s
+  %20s: %s
+  %20s: %s
+  %20s: %s
+  %20s: %s
+  %20s: %s
+  %20s: %s
+`,
+		"Id", service.ID(),
+		"Href", service.HREF(),
+		"Service type", service.Service(),
+		"Service State", service.ServiceState(),
+		"Cluster Name", service.Cluster().Name(),
+		"Created At", service.CreatedAt(),
+		"Updated At", service.UpdatedAt())
```

Comment:
I think the formatting is slightly off on this, I think we should have something similar to describe cluster 
```
╭─croche@localhost ~/Projects/sda/scripts 
╰─$ rosa describe service --id 27R3zVEr2DBnMojF2qPIyspcLuE

                    Id: 27R3zVEr2DBnMojF2qPIyspcLuE
                  Href: /api/service_mgmt/v1/services/27R3zVEr2DBnMojF2qPIyspcLuE
          Service type: ocm-addon-test-operator
         Service State: waiting for cluster
          Cluster Name: croche
            Created At: 2022-04-06 18:22:13 +0000 UTC
            Updated At: 2022-04-06 18:30:03 +0000 UTC
╭─croche@localhost ~/Projects/sda/scripts 
╰─$ rosa describe cluster --cluster croche                
Name:                       croche
ID:                         1rdoojlna7fitub4vg054t3pnmuivtov
External ID:                
OpenShift Version:          
Channel Group:              stable
DNS:                        croche.q6pk.s1.devshift.org
AWS Account:                765374464689
API URL:                    
Console URL:                
Region:                     us-east-1
Multi-AZ:                   false
Nodes:
 - Control plane:           3
 - Infra:                   2
 - Compute:                 2
Network:
 - Service CIDR:            172.30.0.0/16
 - Machine CIDR:            10.0.0.0/16
 - Pod CIDR:                10.128.0.0/14
 - Host Prefix:             /23
STS Role ARN:               arn:aws:iam::765374464689:role/ManagedOpenShift-Installer-Role
Support Role ARN:           arn:aws:iam::765374464689:role/ManagedOpenShift-Support-Role
Instance IAM Roles:
 - Control plane:           arn:aws:iam::765374464689:role/ManagedOpenShift-ControlPlane-Role
 - Worker:                  arn:aws:iam::765374464689:role/ManagedOpenShift-Worker-Role
Operator IAM Roles:
 - arn:aws:iam::765374464689:role/croche-p8o4-openshift-cluster-csi-drivers-ebs-cloud-credentials
 - arn:aws:iam::765374464689:role/croche-p8o4-openshift-cloud-network-config-controller-cloud-cred
 - arn:aws:iam::765374464689:role/croche-p8o4-openshift-machine-api-aws-cloud-credentials
 - arn:aws:iam::765374464689:role/croche-p8o4-openshift-cloud-credential-operator-cloud-credential
 - arn:aws:iam::765374464689:role/croche-p8o4-openshift-image-registry-installer-cloud-credentials
 - arn:aws:iam::765374464689:role/croche-p8o4-openshift-ingress-operator-cloud-credentials
State:                      waiting (Waiting for OIDC configuration)
Private:                    No
Created:                    Apr  6 2022 18:22:11 UTC
Details Page:               https://qaprodauth.cloud.redhat.com/openshift/details/s/27R3zDCAcAaU4HATZrV030Vlp5y
OIDC Endpoint URL:          https://rh-oidc-staging.s3.us-east-1.amazonaws.com/1rdoojlna7fitub4vg054t3pnmuivtov

```

Reply:
Updated: 
```
[rcampos@rh-laptop rosa]$ ./rosa describe service --id 27QUb9SfWsHqHtMNpgnf1xD2A9N
Id:                         27QUb9SfWsHqHtMNpgnf1xD2A9N
Href:                       /api/service_mgmt/v1/services/27QUb9SfWsHqHtMNpgnf1xD2A9N
Service type:               ocs-provider-qe
Service State:              error
Cluster Name:               rc-test
Created At:                 2022-04-06 13:31:11 +0000 UTC
Updated At:                 2022-04-06 18:25:03 +0000 UTC
```